helpers/utils.py

```
@@ -43,7 +47,27 @@
 
-def get_audio_duration(file: Path):
-    """Gets the duration of an audio file in seconds."""
-    duration = AudioSegment.from_file(file).duration_seconds
-    return duration
+def get_audio_duration(file: Path) -> float:
+    """Gets the duration of an audio file in seconds.
+
+    Uses ffprobe to read container metadata only, avoiding decoding the whole file into RAM just for a metric.
+    """
+    try:
+        result = subprocess.run(
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)